src/vehicle.cpp

```diff
@@ -2985,6 +2985,24 @@ bool vehicle::is_moving() const
     return velocity != 0;
 }
 
+bool vehicle::can_use_rails() const
+{
+    bool can_use = true;
+    bool is_wheel_on_rail = false;
+    for( int part_index : wheelcache ) {
```

Comment:
consider caching the rail wheels in vehicle::refresh() so you'd have this instead:
```
    bool can_use = !railcache.empty();
￼     bool is_wheel_on_rail = false;
￼     for( int part_index : railcache ) {

Reply:
So you suggest adding separated cache for rail wheels?

Reply:
yes.